src/clients/entitlements/storage/in-memory/in-memory.cache.ts

```diff
@@ -0,0 +1,224 @@
+import { EntitlementsCache, ExpirationTime, NO_EXPIRE } from '../types';
+import {
+  EntitlementsDto,
+  FeatureBundleDto,
+  FeatureDto,
+  FeatureKey,
+  TenantId,
+  UserId,
+  VendorEntitlementsDto,
+} from '../../types';
+import {
+  ENTITLEMENTS_MAP_KEY,
+  PERMISSIONS_MAP_KEY,
+  SRC_BUNDLES_KEY,
+  getFeatureEntitlementKey,
+} from './in-memory.cache-key.utils';
+import NodeCache = require('node-cache');
+import { pickExpTimestamp } from '../exp-time.utils';
+import { BundlesSource, EntitlementsMap, FeatureSource, PermissionsMap } from './types';
+import { Permission } from '../../../identity/types';
+
+export class InMemoryEntitlementsCache implements EntitlementsCache {
+  private nodeCache: NodeCache;
+
+  private constructor(readonly revision: string) {
+    this.nodeCache = new NodeCache({
+      useClones: false,
+      errorOnMissing: true,
+    });
+  }
+
+  async getFeatureEntitlement(
```

Comment:
the name might be a little misleading, because you are giving back here only the entitlement expiration time if one exist.